### PR TITLE
Npaw Sample App Modified code for external storage aceess

### PR DESCRIPTION
**Pull request by sgupta3112**
Hi @rcvrgs , 

Here I have added AbstractHookActivity class in player package. Also we have added permission in manifest file as well. 

Tested sample app on Android 4.4.2 and Android 6.

### CODE REVIEW

File: NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+			} else {
+				Log.e(TAG, msg);
+			}
+			return;
+		}
+
+		// Automation Hook: to write Notifications to a temporary file on the device/emulator
+		String text = "Notification Received: " + arg1 + " - state: " + player.getState();
+		// Automation Hook: Write the event text along with event count to log file in sdcard if the log file exists
+		log.writeToSdcardLog(text);
```
Comment (brol1dev) [inline NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java]: you missed to use the `writePermission` variable here

File: NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	final String TAG = this.getClass().toString();
+
+	private final SDCardLogcatOoyalaEventsLogger log = new SDCardLogcatOoyalaEventsLogger();
+
+	protected String embedCode;
+	protected String pcode;
+	protected String domain;
+	protected OoyalaPlayerLayoutController playerLayoutController;
+	protected OoyalaPlayerLayout playerLayout;
+	protected PluginOoyala pluginOoyala;
```
Comment (brol1dev) [inline NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java]: I would move `PluginOoyala` to `NPAWDefaultPlayerActivity`.

File: NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/NPAWDefaultPlayerActivity.java [hunk 3]
```diff
@@ -1,141 +1,68 @@
...
-
-  @Override
-  protected void onStop() {
-    super.onStop();
-    Log.d(TAG, "Player Activity Stopped");
-    if (player != null) {
-      player.suspend();
-    }
-    if (isFinishing()) {
-      pluginOoyala.stopMonitoring();
```
Comment (brol1dev) [inline NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/NPAWDefaultPlayerActivity.java]: you got rid of this code to stop and resume `pluginOoyala`, add it again.

**Review by brol1dev — CHANGES_REQUESTED** [review]

File: NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java (hunk 2 above)
Comment (sgupta3112) [reply NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java]: Moved to NPAWDefaultPlayerActivity.

File: NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/NPAWDefaultPlayerActivity.java (hunk 3 above)
Comment (sgupta3112) [reply NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/NPAWDefaultPlayerActivity.java]: Added this in NPAWDefaultPlayerActivity.java file

File: NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java (hunk 1 above)
Comment (brol1dev) [reply NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java]: you haven't updated this.

File: NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/NPAWDefaultPlayerActivity.java [hunk 4]
```diff
@@ -91,51 +70,14 @@ protected void onStop() {
...
   }
 
-  @Override
-  protected void onRestart() {
-    super.onRestart();
-    Log.d(TAG, "Player Activity Restarted");
-    if (player != null) {
-      player.resume();
-    }
-    pluginOoyala.resumeMonitoring();
```
Comment (brol1dev) [inline NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/NPAWDefaultPlayerActivity.java]: i don't see you're resumingMonitoring elsewhere, and you removed this. I think the implementation is broken. Make sure to know when to start, pause, resume and stop monitoring.

**Review by brol1dev — CHANGES_REQUESTED** [review]

File: NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/NPAWDefaultPlayerActivity.java (hunk 4 above)
Comment (sgupta3112) [reply NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/NPAWDefaultPlayerActivity.java]: Yes, We have added this.

File: NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java (hunk 1 above)
Comment (sgupta3112) [reply NPAWSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java]: Updated.

**Review by brol1dev — APPROVED** [review]

**Pull request merged**